### ISSUE TITLE
Feature: `@stream` support

### ISSUE THREAD

**Issue by jpvajda** [apollographql/apollo-client#9710]
One of the disadvantages of GraphQL’s request/response model is that the GraphQL response is not returned to clients until the entire request has finished processing. However, not all requested data may be of equal importance, and in some use cases it may be possible for applications to act on a subset of the requested data. An application can speed up its time-to-interactive if the GraphQL server can send the most important data as soon as it’s ready. The new `@defer` and `@stream` directives allow GraphQL servers to do exactly that by returning multiple payloads from a single GraphQL response. 

Similar to `@defer` the `@stream` directive also allows the client to receive data before the entire result is ready. `@stream` can be used on list fields. Whereas `@defer` can be used  on used on fragment spreads and inline fragments.

### References

https://graphql.org/blog/2020-12-08-improving-latency-with-defer-and-stream-directives/

**Comment by Banou26** [comment]
Is there any ETAs for `@stream`'s support?

Since `@defer` is now on the Apollo Client, I'd be nice if we could also get `@stream`, especially since the Apollo Server already supports both of them.

**Comment by PierrickP** [comment]
It's really hard to find information about current status.

We tried to implement it on our side and it is working "partially".
I made a minimal repository on https://github.com/PierrickP/test-graphql-stream

My `books` list is populated each second :+1: but with `null` :-1: 
![image](https://user-images.githubusercontent.com/1214544/230434919-ffbe273f-9032-43ec-8775-7513e5f8dd2a.png)

If i `curl` the request, it seems to be good

![image](https://user-images.githubusercontent.com/1214544/230435680-cdbc4805-c0d2-4155-ae59-14d0997de780.png)

I'm not 100% sure if i badly implement the resolve or if there are an error somewhere else.
From my understanding, it's on the apollo-client but once again, i'm not sure.
